CalibCalorimetry/HcalTPGAlgos/src/HcaluLUTTPGCoder.cc

```diff
@@ -384,180 +416,262 @@ void HcaluLUTTPGCoder::update(const HcalDbService& conditions) {
   }
 
   for (const auto& id : metadata->getAllChannels()) {
-    if (not(id.det() == DetId::Hcal and topo_->valid(id)))
-      continue;
-
-    HcalDetId cell(id);
-    HcalSubdetector subdet = cell.subdet();
-
-    if (subdet != HcalBarrel and subdet != HcalEndcap and subdet != HcalForward)
-      continue;
-
-    const HcalQIECoder* channelCoder = conditions.getHcalCoder(cell);
-    const HcalQIEShape* shape = conditions.getHcalShape(cell);
-    HcalCoderDb coder(*channelCoder, *shape);
-    const HcalLutMetadatum* meta = metadata->getValues(cell);
-
-    unsigned int bit12_energy =
-        0;  // defaults for energy requirement for bits 12-15 are high / low to avoid FG bit 0-4 being set when not intended
-    unsigned int bit13_energy = 0;
-    unsigned int bit14_energy = 999;
-    unsigned int bit15_energy = 999;
-
-    bool is2018OrLater = topo_->triggerMode() >= HcalTopologyMode::TriggerMode_2018 or
-                         topo_->triggerMode() == HcalTopologyMode::TriggerMode_2018legacy;
-    if (is2018OrLater or topo_->dddConstants()->isPlan1(cell)) {
-      bit12_energy = 16;  // depths 1,2 max energy
-      bit13_energy = 80;  // depths 3+ min energy
-      bit14_energy = 64;  // prompt min energy
-      bit15_energy = 64;  // delayed min energy
-    }
-
-    int lutId = getLUTId(cell);
-    Lut& lut = inputLUT_[lutId];
-    float ped = 0;
-    float gain = 0;
-    uint32_t status = 0;
-
-    if (LUTGenerationMode_) {
-      const HcalCalibrations& calibrations = conditions.getHcalCalibrations(cell);
-      for (auto capId : {0, 1, 2, 3}) {
-        ped += calibrations.effpedestal(capId);
-        gain += calibrations.LUTrespcorrgain(capId);
+    if (id.det() == DetId::Hcal and topo_->valid(id)) {
+      HcalDetId cell(id);
+      HcalSubdetector subdet = cell.subdet();
+
+      if (subdet != HcalBarrel and subdet != HcalEndcap and subdet != HcalForward and subdet != HcalOther)
+        continue;
+
+      const HcalQIECoder* channelCoder = conditions.getHcalCoder(cell);
+      const HcalQIEShape* shape = conditions.getHcalShape(cell);
+      HcalCoderDb coder(*channelCoder, *shape);
+      const HcalLutMetadatum* meta = metadata->getValues(cell);
+
+      unsigned int bit12_energy =
+          0;  // defaults for energy requirement for bits 12-15 are high / low to avoid FG bit 0-4 being set when not intended
```

Comment:
```suggestion
     // defaults for energy requirement for bits 12-15 are high / low to avoid FG bit 0-4 being set when not intended
      unsigned int bit12_energy = 0;
```
looks nicer

Reply:
Sorry for the delay, I will make this change and squeeze the commits

Reply:
@Michael-Krohn before re-launching the tests: since you touched the code anyhow, why not to apply also this comment?